tutorials/single-page-tutorials/nested-nfts-with-wallet-lib.md

```diff
@@ -0,0 +1,423 @@
+{
+  "name": "wallet-setup",
+  "version": "1.0.0",
+  "description": "",
+  "main": "create-mnemonic.js",
+  "dependencies": {
+    "@iota/client": "^3.0.0-alpha.7",
+    "@iota/wallet": "^2.0.2-alpha.21",
+    "bip39": "^3.0.4",
+    "dotenv": "^16.0.1",
+    "ipfs-core": "^0.15.4"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "Apache-2.0"
+}
+```
+
+---
+
+### Prepare Images for your nested NFTs
+
+Add three images to your folder and make sure they are named correctly with `avatar.jpg`, `mask.jpg` and `weapon.jpg`. You can use the images below, or pick your own images as desired:
+
+![Warrior avatar](./images/warrior-avatar.png)
+![Warrior mask](./images/warrior-mask.png)
+![Warrior weapon](./images/warrior-weapon.png)
+
+---
+
+## Scripts
+
+### Create NFTs
+
+Create a new file `create-nfts.js` and add the following code.
+
+:::note
+
+We broke the code into separate snippets to help you understand it better. To make it work, copy all code snippets one after another into the file that you have just created.
+
+:::
+
+#### 1. Imports and parameters
+
+```javascript
+// Libraries
+const { utf8ToHex } = require('@iota/client');
+const { AccountManager } = require('@iota/wallet');
+const IPFS = require('ipfs-core');
+const fs = require('fs');
+
+// Environment variables
+require('dotenv').config({ path: './.env' });
+const password = process.env.SH_PASSWORD;
+const accountName = process.env.ACCOUNT_NAME;
+
+// Network configuration
+const { networkConfig } = require('./networkConfig.js');
+const explorerURL = networkConfig.explorer;
+
+// For better readability, some console output will be printed in a different color
+const consoleColor = '\x1b[36m%s\x1b[0m';
+```
+
+#### 2. Start local IPFS node
+
+The function `startIpfsNode()` starts a local IPFS node for file upload.
+
+```javascript
+// This function starts a local IPFS node for file upload
+async function startIpfsNode() {
+  console.log('\n');
+  console.log(consoleColor, `Start local IPFS node for upload:`);
+
+  let node;
+  if (!node) {
+    node = await IPFS.create({
+      repo: `ipfs_node`,
+    });
+  }
+  return node;
+}
+```
+
+#### 3. Upload file to IPFS
+
+The function `uploadToIPFS()` receives a running ipfs node instance and uploads a file to IPFS from a given file path.
+
+```javascript
+// This function receives a running ipfs node instance and a file path for the upload
+async function uploadToIPFS(ipfsNode, filePath) {
+  try {
+    // Read file from path
+    const file = fs.readFileSync(filePath);
+
+    // Upload file to IPFS
+    const fileAdded = await ipfsNode.add(file);
+    const contentIdentifier = fileAdded.path;
+
+    console.log('\n');
+    console.log(consoleColor, `Your file was successfully uploaded to IPFS:`);
+    console.log(`https://ipfs.io/ipfs/${contentIdentifier}`, '\n');
+
+    return contentIdentifier;
+  } catch (error) {
+    console.error('IPFS upload error', error);
+  }
```

Comment:
I know it is good practice to catch errors and handle them, but if we just print them to the console, I actually don't see any value in the extra code. Just let it crash, which will print the error to the console by default.